### ISSUE TITLE
Slither crashes with assert v == variable (block scoping?)

### ISSUE THREAD

**Issue by gorgos** [crytic/slither#647]
I tried running the tool, but I get the following output:

```
  File "/usr/local/lib/python3.8/site-packages/slither/__main__.py", line 692, in main_impl
    ) = process_all(filename, args, detector_classes, printer_classes)
  File "/usr/local/lib/python3.8/site-packages/slither/__main__.py", line 79, in process_all
    ) = process_single(compilation, args, detector_classes, printer_classes)
  File "/usr/local/lib/python3.8/site-packages/slither/__main__.py", line 62, in process_single
    slither = Slither(target, ast_format=ast, **vars(args))
  File "/usr/local/lib/python3.8/site-packages/slither/slither.py", line 100, in __init__
    self._parser.analyze_contracts()
  File "/usr/local/lib/python3.8/site-packages/slither/solc_parsing/slitherSolc.py", line 353, in analyze_contracts
    self._convert_to_slithir()
  File "/usr/local/lib/python3.8/site-packages/slither/solc_parsing/slitherSolc.py", line 508, in _convert_to_slithir
    contract.convert_expression_to_slithir_ssa()
  File "/usr/local/lib/python3.8/site-packages/slither/core/declarations/contract.py", line 1185, in convert_expression_to_slithir_ssa
    func.generate_slithir_ssa(all_ssa_state_variables_instances)
  File "/usr/local/lib/python3.8/site-packages/slither/core/declarations/function.py", line 1677, in generate_slithir_ssa
    add_ssa_ir(self, all_ssa_state_variables_instances)
  File "/usr/local/lib/python3.8/site-packages/slither/slithir/utils/ssa.py", line 130, in add_ssa_ir
    add_phi_origins(function.entry_point, init_definition, dict())
  File "/usr/local/lib/python3.8/site-packages/slither/slithir/utils/ssa.py", line 541, in add_phi_origins
    add_phi_origins(succ, local_variables_definition, state_variables_definition)
  File "/usr/local/lib/python3.8/site-packages/slither/slithir/utils/ssa.py", line 541, in add_phi_origins
    add_phi_origins(succ, local_variables_definition, state_variables_definition)
  File "/usr/local/lib/python3.8/site-packages/slither/slithir/utils/ssa.py", line 541, in add_phi_origins
    add_phi_origins(succ, local_variables_definition, state_variables_definition)
  [Previous line repeated 11 more times]
  File "/usr/local/lib/python3.8/site-packages/slither/slithir/utils/ssa.py", line 534, in add_phi_origins
    phi_node.add_phi_origin_local_variable(variable, n)
  File "/usr/local/lib/python3.8/site-packages/slither/core/cfg/node.py", line 820, in add_phi_origin_local_variable
    assert v == variable
AssertionError
```

Mac OS: 10.15.6
Python: 3.8.5
slither: 0.6.13
solc: 0.6.12

I can see similar errors in https://github.com/crytic/slither/issues/151 and https://github.com/crytic/slither/issues/534. And it seems to have something to do with block scoping. The contracts do in fact use block scoping in several places to avoid stack too deep errors. Unfortunately it's a closed source project, so I cannot share all code.

But if you have any suggestions, I can look into it more.

**Comment by montyly** [comment]
Hi @gorgos ,

Thank you for reporting this issue. As I don't have access to your source code I assumed the block scoping included yul variables (we actually found a similar bug in another codebase recently). I fixed it in #648, and merged the PR in master.

Let me know if that does not solve your issue.

**Comment by gorgos** [comment]
@montyly Wow that was quick. Yes, it's works now, thank you!
